File: app/src/main/java/com/roblebob/ultradianx/repository/model/AppStateDao.java
package com.roblebob.ultradianx.repository.model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface AppStateDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(AppState appState);

    @Query("SELECT value FROM AppState WHERE :key = `key`")
    LiveData<String> loadValueByKeyLive( String key);

    @Query("SELECT value FROM AppState WHERE :key = `key`")
    String loadValueByKey( String key);
}

File: app/src/main/java/com/roblebob/ultradianx/repository/model/AppStateDao.kt
package com.roblebob.ultradianx.repository.model

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface AppStateDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(appState: AppState)

    @Query("SELECT value FROM AppState WHERE `key` = :key")
    fun loadValueByKeyLive(key: String): LiveData<String>

    @Query("SELECT value FROM AppState WHERE `key` = :key")
    fun loadValueByKey(key: String): String?
}